locust/test/test_web.py

```diff
@@ -345,3 +351,45 @@ def test_index_with_https(self):
         from urllib3.exceptions import InsecureRequestWarning
         requests.packages.urllib3.disable_warnings(category=InsecureRequestWarning)
         self.assertEqual(200, requests.get("https://127.0.0.1:%i/" % self.web_port, verify=False).status_code)
+
+
+class TestWebUIFullHistory(LocustTestCase):
+    STATS_BASE_NAME = "web_test"
+    STATS_FILENAME = "{}_stats.csv".format(STATS_BASE_NAME)
+    STATS_HISTORY_FILENAME = "{}_stats_history.csv".format(STATS_BASE_NAME)
+    STATS_FAILURES_FILENAME = "{}_failures.csv".format(STATS_BASE_NAME)
+
+    def setUp(self):
+        super(TestWebUIFullHistory, self).setUp()
+        self.remove_files_if_exists()
+
+        parser = get_parser(default_config_files=[])
+        self.environment.parsed_options = parser.parse_args(["--csv", self.STATS_BASE_NAME, "--csv-full-history"])
+        self.stats = self.environment.stats
+
+        self.web_ui = self.environment.create_web_ui("127.0.0.1", 0)
+        # Write files now to ensure full-history exists for download - normally done in main
+        locust.stats.write_csv_files(self.environment, self.STATS_BASE_NAME, full_history=True)
+        self.web_ui.app.view_functions["request_stats"].clear_cache()
+        gevent.sleep(0.01)
+        self.web_port = self.web_ui.server.server_port
+
+    def tearDown(self):
+        super(TestWebUIFullHistory, self).tearDown()
+        self.web_ui.stop()
+        self.runner.quit()
+        self.remove_files_if_exists()
+
+    def remove_file_if_exists(self, filename):
+        if os.path.exists(filename):
+            os.remove(filename)
+
+    def remove_files_if_exists(self):
+        self.remove_file_if_exists(self.STATS_FILENAME)
+        self.remove_file_if_exists(self.STATS_HISTORY_FILENAME)
+        self.remove_file_if_exists(self.STATS_FAILURES_FILENAME)
+
+    def test_request_stats_full_history_csv(self):
+        self.stats.log_request("GET", "/test2", 120, 5612)
+        response = requests.get("http://127.0.0.1:%i/stats/requests_full_history/csv" % self.web_port)
+        self.assertEqual(200, response.status_code)
```

Comment:
Maybe validate the actual content of the response as well?

Reply:
Sure, I can add that - it will be somewhat duplicating `test_csv_stats_writer_full_history` - but some duplication in tests are not a bad thing.

Reply:
if it takes a long time to test data data and you do it somewhere else then maybe it is not necessary (but it would be nice)

Reply:
I'll look into it tonight.